docs/admin/settings.rst

```diff
@@ -188,3 +188,97 @@ Engine settings
 
    A few more options are possible, but they are pretty specific to some
    engines, and so won't be described here.
+
+
+.. _settings location:
+
+settings.yml location
+=====================
+
+First, searx will try to load settings.yml from these locations:
+
+1. the full path specified in the ``SEARX_SETTINGS_PATH`` environment variable.
+2. ``/etc/searx/settings.yml``
+
+If these files don't exist (or are empty or can't be read), searx uses the :origin:`searx/settings.yml` file.
+
+.. _ settings use_default_settings:
+
+use_default_settings
+====================
+
+.. note::
+
+   If searx is cloned from a git repository, most probably there is no need to have an user settings.
+
+The user defined settings.yml can relied on the default configuration :origin:`searx/settings.yml` using ``use_default_settings: True``.
+
+In the following example, the actual settings are the default settings defined in :origin:`searx/settings.yml` with the exception of the ``secret_key`` and the ``bind_address``:
+
+.. code-block:: yaml
+
+  use_default_settings: true
+  server:
+      secret_key: "uvys6bRhKHUdFF5CqbJonSDSRN8H0sCBziNSrDGNVdpz7IeZhveVart3yvghoKHA"
+  server:
+      bind_address: "0.0.0.0"
+
+With ``use_default_settings: True``, each settings can be override in a similar way with one exception, the ``engines`` section:
+
+* If the ``engines`` section is not defined in the user settings, searx uses the engines from the default setttings (the above example).
+* If the ``engines`` section is defined then:
+
+   * searx loads only the engines declare in the user setttings.
+   * searx merges the configuration according to the engine name.
+
+In the following example, only three engines are available. Each engine configuration is merged with the default configuration.
+
+.. code-block:: yaml
+
+  use_default_settings: true
+  server:
+      secret_key: "uvys6bRhKHUdFF5CqbJonSDSRN8H0sCBziNSrDGNVdpz7IeZhveVart3yvghoKHA"
+  engines:
+    - name: wikipedia
+    - name: wikidata
+    - name: ddg definitions
+
+Another example where four engines are available. The arch linux wiki engine has a :ref:`token<private engines>`.
+
+.. code-block:: yaml
+
+  use_default_settings: true
+  server:
+      secret_key: "uvys6bRhKHUdFF5CqbJonSDSRN8H0sCBziNSrDGNVdpz7IeZhveVart3yvghoKHA"
+  engines:
+    - name: arch linux wiki
+      tokens: ['$ecretValue']
+    - name: wikipedia
+    - name: wikidata
+    - name: ddg definitions
+
+automatic update
+----------------
+
+The following comand creates or updates a minimal user settings (a secret key is defined if it is not already the case):
+
+.. code-block:: sh
+
+  make SEARX_SETTINGS_PATH=/etc/searx/settings.yml user-settings.update
+
+Set ``SEARX_SETTINGS_PATH`` to your user settings path.
+
+As soon the user settings contains an ``engines`` section, it becomes difficult to keep the engine list updated.
+The following command creates or updates the user settings including the ``engines`` section:
```

Comment:
Do we really need a update tool for the engines? I guess it is hard to patch the engines in the user settings without having a `diff` of the origin engine settings.yml patch (I haven't look right now into the implementation, theses are just my first thoughts).

Reply:
If the administrator want to specify one token for one engine, all the engines have to be written down from this:
```yaml
use_default_settings: true
server:
    secret_key: "uvys6bRhKHUdFF5CqbJonSDSRN8H0sCBziNSrDGNVdpz7IeZhveVart3yvghoKHA"
engines:
  - name: bing
    tokens: ['$ecretValue']
```

...painful.


An alternative approach is to merge the above configuration with the default settings, but in this case it becomes impossible to remove an engine.

Also, may be this tool can be helpful a searx packaged (Debian, Arch, Docker, etc...) is updated : the configuration can be updated automatically with the new engines. Currently the same thing can be done with the full settings.yml but it doesn't seem to be the case of the Debian package for example (see https://github.com/searx/searx/issues/1209 )

Reply:
Another idea:
```yaml
use_default_settings: true
server:
    secret_key: "uvys6bRhKHUdFF5CqbJonSDSRN8H0sCBziNSrDGNVdpz7IeZhveVart3yvghoKHA"
engines:
  - name: bing
    tokens: ['$ecretValue']
remove_engines:  # valid only with use_default_settings: true
  - bing news
  - btdigg
```

In this example, all the engines in the default settings.yml are included except `bing news` and `btdigg`. The `bing` engine has a token.

In this case there is no need to list all the engines, so there is no need to have tool to update the engine list.

Reply:
> In this case there is no need to list all the engines, so there is no need to have tool to update the engine list.

I would really prefer such a solution, making maintenance less complicated and from searx-admin's view it is easy to understand.  Thanks!